### DISCUSSION TITLE
illogical-impulse on NixOS

### DISCUSSION THREAD

**Discussion by RipDr19** [end-4/dots-hyprland#1093] [Extra Distros]
- **I have read the instructions**
---
## The issue
- Linux distro: Nix OS
- I am trying to manual install this dot in my nix os setup but can clone the .config folder directly and rsync also doesn't seem to help at all

How can I clone the config and local folder of this repo directly to just get the dots which I can then manually paste to my .config folder?

**Comment by end-4** [comment]
EDIT: Configs to look at:
- https://github.com/xBLACKICEx/end-4-dots-hyprland-nixos


_Below is my initial comment while I was figuring out how to use my dotfiles on NixOS. I do realize this isn't the proper way to do things on that OS. Scroll way down for more helpful comments_

1. Clone then `git checkout illogical-impulse`
2. See the command that checks for existence of `pacman` in `install.sh` at line 8? Just comment that out with a `#` then run the script. Skip package installation stuff (for now you have to do that manually in your nix config) and say yes when prompted to copy files. 
3. Beware of conflicts with home-manager and you should be fine

**Comment by clsty** [comment]
That's interesting, I thought NixOS has a total different way dealing with configs... seems like it's actually alike normal Linux distros at least for `~/.config`.

In that case, maybe it's not difficult to have a try on NixOS for me.
And if it can be done, I'll try to implement a install script for NixOS some other time.

**Comment by RipDr19** [comment]
Well it does handles differently too, if you are planning to use flakes or home managers (basically trying to automate it) but otherwise the fallback method still exists

**Comment by end-4** [comment]
It's going to be different if you use home-manager. What it does is generate contents in home folder based on configurations in Nix format.
  - Specifically for AGS, there's an option to **not** have home-manager manage `~/.config/ags`. I'm having it like this since folders generated by home-manager are read-only, which is something that makes my config not work at present.

Btw I can't figure out how to make GtkSourceView work (which is required for left sidebar) :sob:

**Comment by RipDr19** [comment]
And here I am all done and up just I can't seem to understand how to install ags in nix

**Comment by end-4** [comment]
> And here I am all done and up just I can't seem to understand how to install ags in nix

`flake.nix` inputs:
```nix
ags.url = "github:Aylur/ags";
```

then you can source a file like this to `home.nix`
https://github.com/end-4/CirnOS/blob/main/home-manager/ags.nix

**Comment by clsty** [comment]
Will be better if the git checkout (commit hash) of ags repo could be specified.

**Comment by end-4** [comment]
there must be a way... that's why i tried to use nixos in the first place
using `ref` in inputs of flake.nix i think

**Comment by RipDr19** [comment]
I still can't understand flakes at all, I will shift back to arch for now, do you think this setup can work on void linux though?

**Comment by end-4** [comment]
try it and see

**Comment by RipDr19** [comment]
I tried from the ags wiki and it returned that flake path is not set with a bunch of other things

**Comment by solitudealma** [comment]
> I tried from the ags wiki and it returned that flake path is not set with a bunch of other things

i think you should find a config for nixos that looks good structure. like: [zaneyos](https://gitlab.com/Zaney/zaneyos). and then, you can change it to what you like

**Comment by celesrenata** [comment]
I had a stab at it a few months back, I'm going through and updating for 24.05 this weekend.

Added a feature to adjust foot transparency on the fly from the effects menu. :)

https://github.com/celesrenata/nix-flakes

**Comment by Makrennel** [comment]
I've gone ahead and taken a stab at making a [home manager module and flake](https://github.com/Makrennel/dots-hyprland/tree/nixos) for this with a config to allow users to exclude certain files from being sourced in the config so they can be sourced by the user manually if modifications are necessary.

It's probably not a very nix-like way to do things, but I've tried to keep it such that it sticks to the config files in this repository rather than rewriting everything in nix, and it can just be added like any other flake input rather than having to be installed like a whole new system configuration.

That said, I haven't implemented the installation of any required packages yet besides the pinned ags revision, that'll be the next step.

I may also have to figure out some things to allow the auto-generation components to work properly – or maybe use some kind of config to force declarative generation rather than imperative.

**Comment by Makrennel** [comment]
> That said, I haven't implemented the installation of any required packages yet besides the pinned ags revision, that'll be the next step.

So I've managed to get it running such that it will, upon being enabled, install the correct packages and Hyprland can be started. There seems to be an issue with getting ags to work though. I not really sure what's causing it, and ags is capable of displaying a default config, but not the one provided in illogical-impulse.

Here is the log from when running `ags` up to killing it with `ags -q`

```
/nix/store/4arj1mz42giyrpqxiir1nvllj1hg708n-gvfs-1.54.2/lib/gvfs/libgvfscommon.so: undefined symbol: g_once_init_leave_pointer
Failed to load module: /nix/store/4arj1mz42giyrpqxiir1nvllj1hg708n-gvfs-1.54.2/lib/gio/modules/libgvfsdbus.so

(com.github.Aylur.ags:8744): Gjs-WARNING **: 01:17:57.454: JS ERROR: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data @ file:///home/alice/Local/etc/ags/modules/cheatsheet/keybinds.js:21:17
No /dev/i2c devices exist.
ddcutil requires module i2c-dev.

(com.github.Aylur.ags:8744): Gjs-CRITICAL **: 01:26:57.284: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
The offending signal was destroy on Ags_Stack 0x10f0510.
== Stack trace for context 0x7b4c80 ==

(com.github.Aylur.ags:8744): Gjs-CRITICAL **: 01:26:57.284: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
The offending signal was destroy on Ags_Stack 0x10f0510.
== Stack trace for context 0x7b4c80 ==

(com.github.Aylur.ags:8744): Gjs-CRITICAL **: 01:26:57.284: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
The offending signal was destroy on Ags_Label 0x10e3e50.
== Stack trace for context 0x7b4c80 ==

(com.github.Aylur.ags:8744): Gjs-CRITICAL **: 01:26:57.284: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
The offending signal was destroy on Ags_Label 0x10ef5e0.
== Stack trace for context 0x7b4c80 ==
```

The `CRITICAL` part of the log only prints when ags is killed.
From this maybe it's a problem with gvfs? I'm not really sure.

**Comment by bigsaltyfishes** [comment]
I'm working on this too. I packed it into a home manager module (based on CirnOS, I rewrite most of the dotfiles with Nix, and with some of my own customizations.), though there may still some features not working (Maybe there are still some deps I ignored). There is a weird cache files permission created by ags (Permission of files in ~/.cache/ags/user/generated/ is all set as 444, I think it should be 644 since ags need to update them when changing wallpapers.). You may need to set them to 644 manually after the first time you set wallpaper.

https://github.com/bigsaltyfishes/end-4-dots

**Comment by clsty** [comment]
Note: executable name switched from `ags` to `agsv1` for preparing possible migration to agsv2 (#1092).

On typical Linux system you just need to run `sudo mv /usr/local/bin/ags{,v1}`, but I don't know what to do for NixOS.

**Comment by clsty** [comment]
Note: for python packages, `gradience` and its deps `blueprint-compiler` `material-color-utilities-python` `pillow==9.5.0` has been removed since #1273 . No need to install them anymore.

**Comment by HynDuf** [comment]
Is there any updates on this?

**Comment by sitolam** [comment]
Yeah, I would also like to see an update on this, I used [bigsaltyfishes](https://github.com/bigsaltyfishes) configuration for the ags version of this repo, but I would really like to see a nixos version of the quickshell version. [bigsaltyfishes](https://github.com/bigsaltyfishes) take on this is also far from fully nixos, but it worked ig.

**Comment by celesrenata** [comment]
Ive kinda cobbled something together for quickshell.
https://github.com/celesrenata/end-4-flakes

**Reply by sitolam** [reply]
Looks cool, I think it is this link though: https://github.com/celesrenata/end-4-flakes. Could you maybe sent me a video of the ui and maybe you should put some screenshots on the README. Love it though, will take a look

**Reply by Gresmond** [reply]
This is phenomenal work. I hope you can spread the good word.

**Reply by sitolam** [reply]
This repo from @celesrenata also look really cool: https://github.com/celesrenata/nix-flakes. I just would need a little bit more help to all set it up I think, but looks really cool, but don't really know how to do it all.

**Reply by celesrenata** [reply]
Here is an example of my current build as video:

* https://s3.us-east-1.amazonaws.com/celestium.net/videos/recording_2025-09-18_18.22.53.mp4
* https://s3.us-east-1.amazonaws.com/celestium.net/videos/recording_2025-09-18_18.36.40.mp4

Here is the latest release: https://github.com/celesrenata/nix-flakes

**Reply by sitolam** [reply]
> Here is an example of my current build as video:
> 
>     * https://s3.us-east-1.amazonaws.com/celestium.net/videos/recording_2025-09-18_18.22.53.mp4
> 
>     * https://s3.us-east-1.amazonaws.com/celestium.net/videos/recording_2025-09-18_18.36.40.mp4
> 
> 
> Here is the latest release: https://github.com/celesrenata/nix-flakes

Look really cool, just noticing that the notifications aren't themed. But looks really cool

**Reply by celesrenata** [reply]
I added support for foot to act exactly like iterm as well as blur/transparency support for it in the main settings menu.

**Reply by celesrenata** [reply]
Notifications are themed without the Quickshell reset ctrl+alt+R based on default keybinds. But it still suffers from hangups that I had from AGS1, they seem to appear mostly with Tidal until the reset. There is a lot of room for improvement, but it does mostly work, feel free to fork or run a pull request as needed. It is enough to build my desktop and laptop flakes off of, but it could probably be improved!

**Comment by soymou** [comment]
I've made a complete NixOS flake that handles the full installation. QuickShell, Python environment, dynamic theming, and config persistence all work out of the box.
Repository: https://github.com/soymou/illogical-flake

**Comment by bnuy1** [comment]
I have been porting the config over to my own interpretation of these dotfiles: 
https://github.com/bnuy1/bnuynix

i added a few features and removed alot on the way into making this my own :/ 
i still have a long way to go but i got quite a bit done

**Comment by celesrenata** [comment]
I refreshed my upstream, and added more AI features (they still need a little polishing), added documentation. It is all vibe coded, but I daily it happily

https://github.com/celesrenata/end-4-flakes/tree/upstream-sync-2026
